### PR TITLE
pbench-rsync-satellite: Fix tarball deletions and send status email

### PR DESCRIPTION
**Pull request by ndokos**
Tarball deletion was done for every tarball that was ever rsynced
from the remote: that's incorrect.

Check MD5 and delete tarballs that came during the current run only.

Send status email containing:
- tarball removal errors on the satellite server
- MD5 check failures
- Unfinished business from process-work-queue (if any).

Reorganize the structure of the log directory:

   logs/pbench-rsync-satellite/$PROG.error
                               $PROG.log
                               <prefix>/run-<TS>/$PROG.wq
                                                 $PROG.wq.current-batch
                                                 $PROG.wq.remaining
                                                 <host>/rsync.log
                                                        md5-checks.log
                                                        rm.log
                                                 ...

Fix comments to reflect current reality.

### CODE REVIEW

**Comment by ndokos** [comment]
RFC - this still needs testing and unit tests before merging.
@portante: can you please review?

File: server/pbench/bin/pbench-rsync-satellite [hunk 1]
```diff
@@ -66,64 +88,136 @@ WONT-INDEX*
...
+# directory of the corresponding host. We let pbench-unpack-tarballs
+# take care of the rest. process-work-queue waits for the last one in
+# the current batch to finish and then submits another batch. A batch
+# consists of a number of tarballs with a total size < 50MB - if the
+# constraint cannot be satisfied, a batch consists of a single tarball.
+# That way, we make progress through the queue until we are done.
+
+# XXX - limitations:
+# DUPLICATE names cause problems - skip them
+# No subdirectories - there is is a badly named (the name includes a '*' and that causes
```
Comment (portante) [inline server/pbench/bin/pbench-rsync-satellite]: s/is is/is/

File: server/pbench/bin/pbench-rsync-satellite-process-work-queue [hunk 2]
```diff
@@ -61,7 +60,10 @@ def process(todo, options):
         if options.verbose:
             print("os.symlink({}, {})".format(os.path.join(x.archive, x.name), x.link_name))
         if not options.dryrun:
-            os.symlink(os.path.join(x.archive, x.name), x.link_name)
+            try:
+                os.symlink(os.path.join(x.archive, x.name), x.link_name)
+            except OSError:
```
Comment (portante) [inline server/pbench/bin/pbench-rsync-satellite-process-work-queue]: Should we report the error encountered?

Comment (ndokos) [inline server/pbench/bin/pbench-rsync-satellite-process-work-queue]: Any tarball that triggers an error here is added to the "skipped" list, which is dumped at the end and reported in the status message. I was assuming that we would go back and investigate such failures. The common
case should be that the link exists for some reason: the script could probably deal with that case, but first I'd like to find out how common it is. So I'd like to run it as is for the time being and tweak the handling on the basis of the actual errors we get. Does that seem OK to you?

**Pull request merged**